lib/Common/Memory/Recycler.cpp

```diff
@@ -947,12 +947,12 @@ Recycler::IsSweeping() const
 }
 
 void
-Recycler::SetIsScriptActive(bool isScriptActive)
+Recycler::SetIsScriptActive(bool isScriptActive, bool doCleanup)
 {
     Assert(this->isInScript);
     Assert(this->isScriptActive != isScriptActive);
     this->isScriptActive = isScriptActive;
-    if (isScriptActive)
+    if (isScriptActive && doCleanup)
```

Comment:
No. We just don't want to keep calling the same sub interface unnecessarily. If there is something to dispose etc. we better do it. We were keep updating tick threshold to a future value + all the sub calls.